[GlennOngJunJie] iP

GlennOngJunJie opened the pull request:
# Glenn's DukePro
> "A goal without a plan is just a wish" - Larry Elder

Glenn's DukePro only has 3 prerequisites:

- Knows English
- Have things to do
- Not be **LAZY**

All you need to do to set it up:

1.  Download it from [here](https://github.com/GlennOngJunJie/ip/releases/tag/A-Jar)
2. Choose where you want to save your progress on your desktop
3. Add todos, events, deadlines
4. Enjoy using :smile:

Things you can do in the list:

- [x] todo Study
- [x] deadline Submit proposal by 6 Feb
- [x] event Movie Screening from 7 Feb 7pm to 9pm

To run the program make use of the `.run()`:
```Java
public static void main(String[] args) throws DukeException, IO Exception {
          new Duke(data/GlennTasks.txt).run();
}
```

Commit:
Add Gradle support

GlennOngJunJie committed:
Duke Level 1

GlennOngJunJie committed:
Level 2 implemented

GlennOngJunJie committed:
Level 3 done

GlennOngJunJie committed:
level 4 done

GlennOngJunJie committed:
TextUiTesting done

GlennOngJunJie committed:
Level 5 done

GlennOngJunJie committed:
Level 6 done

GlennOngJunJie committed:
Add Level 7

GlennOngJunJie committed:
Add Level 8

GlennOngJunJie committed:
A-MoreOOP Done

GlennOngJunJie committed:
A-Packages Done

GlennOngJunJie committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

GlennOngJunJie committed:
A-JUnit Done

GlennOngJunJie committed:
A-JavaDoc Done

GlennOngJunJie committed:
A-JavaDoc Done

GlennOngJunJie committed:
Revert "A-JavaDoc Done"

This reverts commit 11cc6bec6dc2087f8cc6d707d2370ed741328a8f.

GlennOngJunJie committed:
A-CodingStandard Done

GlennOngJunJie committed:
Level-9 Done

GlennOngJunJie committed:
Resolved Merge Conflict

GlennOngJunJie committed:
Merge Conflict Level 9 Done

4ndrelim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,42 @@
+import duke.*;
```
I understand you only have 1 overall package - duke. But perhaps instead of importing the whole package, list only the required classes for this file?

4ndrelim commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+package duke;
```
Nice segregation of import statements!

4ndrelim commented on src/main/java/duke/Tasklist.java:
```diff
@@ -0,0 +1,114 @@
+        case "event":
+            int position1 = content.indexOf("/from ");
+            int position2 = content.indexOf("/to ");
+            Event eventTask = new Event(content.substring(6, position1), content, content.substring(position1 + 6, position2), content.substring(position2 + 4));
```
I noticed the line length here (and elsewhere) may be slightly too long. Perhaps you can add line breaks and have wrapped lines instead to improve readability?

4ndrelim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,42 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private Storage storage;
+    private Tasklist tasks;
```
Just a minor naming convention - task list is actually 2 separate words so perhaps pascal case would be more appropriate here

4ndrelim commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,65 @@
+package duke;
+
+public class Task {
```
Perhaps a high-level javadocs overview of the main functionalities of each public class could be added so users know what the class is primarily used for without having to delve into its methods.

4ndrelim reviewed (commented):
Overall, great job in following the coding standard! 👍 Your code is neat and relatively easy to parse and read. All the best!

cylithegreat commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,42 @@
+    private Tasklist tasks;
+    private Ui ui;
+
+    public Duke(String filepath) throws IOException {
```
Perhaps the filepath variable should be named as filePath following the camel case convention?

cylithegreat commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,61 @@
+            if (userInput.substring(9).equals("")) {
+                throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+            } else {
+                tasklist.addingActivities("deadline", userInput);
```
Any reason you use addingActivities rather than addActivities here? Perhaps it would be good to standardize to present simple tense as indicated here https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/topics.html#w4-6-code-quality-naming?

cylithegreat commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+    private String filePath;
+    private PrintWriter storeInFile;
+    private boolean filePresent = false;
+    private File fileCreation;
```
I'm not sure if I like how this variable is named, as it feels not very intuitive and self explanatory to me.

cylithegreat commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+public class Storage {
+    private String filePath;
+    private PrintWriter storeInFile;
+    private boolean filePresent = false;
```
Perhaps this could be renamed as isFilePresent following the convension for naming bool variables?

cylithegreat commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,111 @@
+     * @throws FileNotFoundException if the location to store the tasklist in is not available.
+     */
+    public void saveToFile(Tasklist tasklist) throws FileNotFoundException {
+        storeInFile = new PrintWriter(fileCreation);
```
Should storeInFile be named in a way that more explicitly and intuitively conveys its function?

cylithegreat commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,65 @@
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Nice use of variable name "isDone" rather than "done" for the bool variable name👍 It was a common mistake.

cylithegreat reviewed (commented):
I like how neat your code is! It is very readable and defintely easily understandable.
Just some minor recommendations regarding the naming conventions.
Overall great work 👍

GlennOngJunJie committed:
A-CheckStyle Added

GlennOngJunJie committed:
GUI done

GlennOngJunJie committed:
Code assumes that mark,delete, unmark all have valid numbers

Code will run into error if user does not provide a number or a number greater than current task list

Assertion are added and new return statement added if such a scenario arises

This way the programme automatically cuts and would not run further

GlennOngJunJie committed:
Merge pull request #2 from GlennOngJunJie/branch-A-Assertions

Code assumes that mark,delete, unmark all have valid numbers

GlennOngJunJie committed:
Alot of dead code that has not been deleted

The code looks messy and unreadable

Remove all unnecessary code and imports that are not used

Everything will look neater and flow better

GlennOngJunJie committed:
Merge pull request #3 from GlennOngJunJie/branch-A-CodeQuality

Alot of dead code that has not been deleted

GlennOngJunJie committed:
Version does not provide a user guide

It is difficult for first time user to know the syntax of how to use the program

User guide and examples added

Simple and intuitive way of asking for help and examples will cater to all ages

GlennOngJunJie committed:
UI screenshot not updated, assertions does not account for arraylist size equals marked task

Prevent errors if arraylist size equals to task number

Added greater than or equals to

GlennOngJunJie committed:
No updated user guide

User guide included to show user what this programme is about and to get them started

User guide included with all the set up details, how to use, what it can do and examples

Clear and easy to find

GlennOngJunJie committed:
Updated with updated JAR file link

GlennOngJunJie committed:
Update to checkstyle and ParserTest

GlennOngJunJie committed:
Updates to Gradle file to launch launcher

GlennOngJunJie committed:
Dummy push to fix link

GlennOngJunJie committed:
Fixed JAR file link

GlennOngJunJie committed:
Fixed bug where assert was on the wrong line and delete and unmark with wrong values

GlennOngJunJie committed:
More debugging done

GlennOngJunJie committed:
Adding readme to /docs filed

GlennOngJunJie committed:
Merge branch 'Branch-Level-10'

* Branch-Level-10:
  Adding readme to /docs filed
